packages/create-mechanic/function-examples/adaptive-grid/function/index.js

```diff
@@ -0,0 +1,411 @@
+      >
+        {title.toUpperCase()}
+      </text>
+    </svg>
+  );
+};
+
+export const inputs = {
+  width: {
+    type: "number",
+    default: 1000
+  },
+  height: {
+    type: "number",
+    default: 1000
+  },
+  randomRatio: {
+    type: "groupToggle",
+    default: true,
+    label: "Random Ratio",
+    inputs: {
+      columnOneRatio: {
+        type: "number",
+        slider: true,
+        default: 2,
+        min: 1,
+        max: 10,
+        step: 1
+      },
+      columnTwoRatio: {
+        type: "number",
+        slider: true,
+        default: 2,
+        min: 1,
+        max: 10,
+        step: 1
+      },
+      columnThreeRatio: {
+        type: "number",
+        slider: true,
+        default: 1,
+        min: 1,
+        max: 10,
+        step: 1
+      },
+      border: {
+        type: "number",
+        slider: true,
+        default: 15,
+        min: 1,
+        max: 100,
+        step: 1
+      },
+      gutter: {
+        type: "number",
+        slider: true,
+        default: 15,
+        min: 1,
+        max: 100,
+        step: 1
+      },
+      imageColumn: {
+        type: "number",
+        slider: true,
+        default: 1,
+        min: 1,
+        max: 3,
+        step: 1
+      }
+    }
+  },
+  randomColor: {
+    type: "groupToggle",
+    default: false,
+    label: "Random Color",
+    inputs: {
+      backgroundColor: {
+        type: "color",
+        model: "hex",
+        default: "#000000"
+      },
+      textColor: {
+        type: "color",
+        model: "hex",
+        default: "#ffffff"
+      },
+      titleColor: {
+        type: "color",
+        model: "hex",
+        default: "#E94825"
+      }
+    }
+  },
+  image: {
+    type: "image",
+    multiple: false
+  },
+  filterOpacity: {
+    type: "number",
+    default: 20,
+    min: 0,
+    max: 100,
+    step: 1,
+    slider: true
+  },
+  textSize: {
+    type: "number",
+    default: 20
+  },
+  titleSizeAdjust: {
+    type: "number",
+    slider: true,
+    default: 0,
+    min: -5,
+    max: 15,
+    step: 1
+  },
+  title: {
+    type: "text",
+    default: "Mechanic"
+  },
+  textOne: {
+    type: "text",
+    default: "MUNUS SHIH"
+  },
+  textTwo: {
+    type: "text",
+    default: "MECHANIC.DESIGN INFO@MECHANIC.DESIGN"
+  },
+  textThree: {
+    type: "text",
+    default: "@MECHANIC 781 12TH ST, 8A, NEW YORK, NY 10003"
+  },
+  textFour: {
+    type: "text",
+    default: "MUNUS@MECHANIC.DESIGN"
+  },
+  grid: {
+    type: "boolean",
+    default: false
+  }
+};
+
+export const presets = {
+  "Instagram Story": {
+    width: 1080,
+    height: 1920
+  },
+  "Instagram Post": {
+    width: 1080,
+    height: 1080
+  },
+  Poster: {
+    width: 812,
+    height: 1148
+  },
+  Banner: {
+    width: 1640,
+    height: 624
+  },
+  Ticket: {
+    width: 394,
+    height: 1126
+  }
+};
+
+export const settings = {
+  engine: require("@mechanic-design/engine-react"),
+  showMultipleExports: true
```

Comment:
When exporting this as a PNG the image that's rendered in the preview by default doesn't show up. When I upload a custom image it works.
![grid-2025-05-06-21-09 (1)](https://github.com/user-attachments/assets/98640370-ad92-4db9-b1f5-378999c105ff)
![grid-2025-05-06-21-09](https://github.com/user-attachments/assets/fb667009-99c5-472f-b6c0-4ca5feb83c39)

Reply:
Thanks URL loading. Switched it to load the image directly.